src/raw/client.rs

```diff
@@ -563,3 +628,117 @@ impl Client {
         self.atomic.then(|| ()).ok_or(Error::UnsupportedMode)
     }
 }
+
+#[derive(Clone)]
+pub struct SyncClient {
+    client: Client,
+}
+
+impl SyncClient {
+    /// The Sync version of Client
```

Comment:
The comments should be moved to the definition of the structure.
```suggestion
    /// The synchronous version of RawClient
```